src/blockdata/block.rs

```diff
@@ -159,6 +160,58 @@ impl BlockHeader {
     }
 }
 
+/// Bitcoin block version number
+///
+/// Originally used as a protocol version, but repurposed for soft-fork signaling.
+///
+/// ### Relevant BIPs
+///
+/// * [BIP9 - Version bits with timeout and delay](https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki) (current usage)
+/// * [BIP34 - Block v2, Height in Coinbase](https://github.com/bitcoin/bips/blob/master/bip-0034.mediawiki)
+#[derive(Copy, PartialEq, Eq, Clone, Debug, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct BlockVersion(pub i32);
+
+impl BlockVersion {
+    /// BIP-9 compatible version number that does not signal for any softforks.
+    pub const NO_SOFT_FORK_SIGNALLING: Self = Self(0x20000000);
+    /// BIP-9 soft fork signal bits mask.
+    const BIP_9_BITS_MASK: u32 = 0x1FFFFFFF;
+    /// BIP-9 enable mask.
+    const BIP_9_ENABLE_MASK: u32 = 0xE0000000;
+
+    /// Check whether the version number is signalling a soft fork at the given bit.
+    ///
+    /// A block is signalling for a soft fork under BIP-9 if the first 3 bits are `001` and
+    /// the version bits for the specific soft fork are toggled on.
+    pub fn is_signalling_soft_fork(&self, bit: u8) -> bool {
+        if bit > 29 {
+            return false;
+        }
+
+        (self.0 as u32 & Self::BIP_9_ENABLE_MASK) >> 29 == 1 && (self.0 as u32 & Self::BIP_9_BITS_MASK) & 1 << bit > 0
+    }
+}
```

Comment:
While playing with this PR I came up with (feel free to use any, all, or none)
```rust
impl BlockVersion {
    /// BIP-9 compatible version bits value that does not signal for any softforks.
    pub const VERSION_BITS: Self = Self(Self::USE_VERSION_BITS as i32);
    /// BIP-9 soft fork signal bits mask.
    const VERSION_BITS_MASK: u32 = 0x1FFFFFFF;
    /// If using version bits the first three bits must be `001`.
    const USE_VERSION_BITS: u32 = 0x2000_0000;
    
    
    /// Check whether the version number is signalling a softfork at the given bit.
    ///
    /// A block is signalling for a softfork under BIP-9 if the first 3 bits are `001` and
    /// the version bit for the specific soft fork is toggled on.
    pub fn is_signalling_soft_fork(&self, bit: u8) -> bool {
        // Only bits [0, 28] inclusive are used for signalling.
        if bit > 28 {
            return false;
        }

        // To signal using version bits the top three bits must be 001
        if (self.0 as u32) & !Self::VERSION_BITS_MASK != Self::USE_VERSION_BITS {
            return false;
        }

        // Finally, the bit is set if signalling soft fork.
        (self.0 as u32 & Self::VERSION_BITS_MASK) & (1 << bit) > 0
    }
}
```

Reply:
Thanks, makes it easier to read by a tonne.